### DISCUSSION TITLE
pass email and password and get the token

### DISCUSSION THREAD

**Discussion by quantrpeter** [microsoftgraph/msgraph-sdk-python#840] [Q&A] [answered]
Hi.
   possible to pass email and password and get the token? instead of typing the code in a webpage
thanks
Peter

**Comment by andrueastman** [comment, marked as answer]
@quantrpeter Any chance you've tried using the UsernamePasswordCredential class? See here for reference https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.usernamepasswordcredential?view=azure-python

**Reply by quantrpeter** [reply]
thank you hero
